keanu-project/src/test/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensorTest.java

```diff
@@ -312,4 +312,35 @@ public void canCalculateProductOfVector() {
         assertTrue(vectorA.isVector() && vectorB.isVector());
     }
 
+    @Test
+    public void scalarMinusInPlaceTensorBehavesSameAsMinus() {
+        DoubleTensor scalar = DoubleTensor.scalar(1);
+        DoubleTensor tensor = DoubleTensor.create(2, new int[] {1, 4});
+
+        assertArrayEquals(scalar.minus(tensor).asFlatDoubleArray(), scalar.minusInPlace(tensor).asFlatDoubleArray(), 1e-6);
+    }
+
+    @Test
+    public void scalarPlusInPlaceTensorBehavesSameAsPlus() {
+        DoubleTensor scalar = DoubleTensor.scalar(1);
+        DoubleTensor tensor = DoubleTensor.create(2, new int[] {1, 4});
+
+        assertArrayEquals(scalar.plus(tensor).asFlatDoubleArray(), scalar.plusInPlace(tensor).asFlatDoubleArray(), 1e-6);
+    }
+
+    @Test
+    public void scalarTimesInPlaceTensorBehavesSameAsTimes() {
+        DoubleTensor scalar = DoubleTensor.scalar(1);
+        DoubleTensor tensor = DoubleTensor.create(2, new int[] {1, 4});
+
+        assertArrayEquals(scalar.times(tensor).asFlatDoubleArray(), scalar.timesInPlace(tensor).asFlatDoubleArray(), 1e-6);
+    }
+
+    @Test
+    public void scalarDivInPlaceTensorBehavesSameAsDiv() {
+        DoubleTensor scalar = DoubleTensor.scalar(1);
+        DoubleTensor tensor = DoubleTensor.create(2, new int[] {1, 4});
+
+        assertArrayEquals(scalar.div(tensor).asFlatDoubleArray(), scalar.divInPlace(tensor).asFlatDoubleArray(), 1e-6);
```

Comment:
Given the numbers in use here, we should be fine not using a delta - it would be quite bad if a single op on a small whole number caused 1e-6 error - especially if we're using Double precision values.

Reply:
The delta is required for comparisons on doubles.

Reply:
Aha - thanks, I see this in the API now (the version without delta is deprecated, but does exist).  I'm still a bit nervous about such a "large" delta.  I guess it's not the end of the world though!